joblib/memory.py

```diff
@@ -540,8 +542,8 @@ def _write_func_code(self, filename, func_code, first_line):
         # sometimes have several functions named the same way in a
         # file. This is bad practice, but joblib should be robust to bad
         # practice.
-        func_code = '%s %i\n%s' % (FIRST_LINE_TEXT, first_line, func_code)
-        with open(filename, 'w') as out:
+        func_code = u'%s %i\n%s' % (FIRST_LINE_TEXT, first_line, func_code)
+        with open(filename, 'w', encoding="UTF-8") as out:
```

Comment:
Maybe it's just me but I got a bit confused how this line could even work for python 2.7 since open doesn't have a 'encoding' parameter.

Is it not preferrable to do `import io` and use `io.open` here rather than `from io import open`?